### PR TITLE
Update DG implementation for UI and show command

### LINKED ISSUE #227: Update implementation section of developers guide

### PR DESCRIPTION
**Pull request by Jayne1010**
Closes #227 
Some issues:

-  Unable to make the class diagram for UI implantation (apparently all my class diagrams were unable to show up for some reason)
- Design Consideration & Alternatives Considered different for the rest of the commands as the rest of commands were all repetitive 
- If so, maybe can do with @jiahui0309 to update the rest of the diagram

Following is the sequence diagram activity for the show command:

Screenshot 2024-04-14 at 9 43 47 PM

Screenshot 2024-04-14 at 1 31 19 AM

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -411,6 +411,100 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+#### Design Consideration & Alternatives Considered 
+
+**Aspect: How the notes of the patient are displayed**
+
+- **Alternative 1 (Current Choice):** Display the notes of the patients in a separate panel.
+    - Pros: The information of patient is more well organised. It allows users to have a more comprehensive view of their patient notes.
+  
+    - Cons: One additional command is needed to view the notes of the patients.
+
+- **Alternative 2:** Display the notes of the patients in the `PersonalListCard` with the rest the of the details.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: do u mean person card or person list panel? 
we can standardise whether we want to code block for it as we didn't code block for patient list panel above :)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -411,6 +411,100 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+    - Pros: The user does not need to enter an extra command to view the notes of the patients.
+  
+    - Cons: It is very difficult to view the notes of the individual patient as there is too much information displayed on the `PersonalListCard`.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: same as above

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -411,6 +411,100 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+- **Alternative 2:** Clears patient's notes on Patient Note Display with another command.
+
+    - Pros: The command is more intuitive.
+
+    - Cons: Users are required to know an extra command, which reduce usability of application.
+
+### NoteDisplay (UI Component)
+
+The `NoteDisplay` allows the users to view the notes of the selected patient in the patient list. All notes in the `Person Card`.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: can we consider "All notes are displayed in the Person Card"?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -411,6 +411,100 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+    <puml src="diagrams/ShowCommandActivityDiagram.puml" width="250"/>
+</div>
+
+#### Design Considerations & Alternatives Considered 
+
+**Aspect: How to show the notes of patient on Patient Note Display.**
+
+- **Alternative 1 (Current Choice):** Show patient's notes on one command using `IC_NUMBER`.
+
+    - Pros: Users able to view specific user's notes.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: can we consider "Users are able..."?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -411,6 +411,100 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+    - Cons: Only allow for one parameter which is `IC_NUMBER`.
+
+- **Alternative 2:** Only show the patient notes which was last searched for.
+    - Pros: Easy to implement.
+
+    - Cons: Brings inconvenient to users if they only want to view a specific user's notes.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: can we change to "inconvenience" instead of "inconvenient"? :)

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -411,6 +411,100 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+- **Alternative 1 (Current Choice):** Clears patient's notes on Patient Note Display using one command.
+    - Pros: Easy to implement. Users do not need to remember so many commands.
+
+    - Cons: The command might not be intuitive.
+
+- **Alternative 2:** Clears patient's notes on Patient Note Display with another command.
+
+    - Pros: The command is more intuitive.
+
+    - Cons: Users are required to know an extra command, which reduce usability of application.
```
Comment (jiahui0309) [inline docs/DeveloperGuide.md]: can we consider "reduces" instead of "reduce"?

**Review by jiahui0309 — CHANGES_REQUESTED** [review]
Some minor comments on the content!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Jayne1010) [reply docs/DeveloperGuide.md]: okok

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -371,7 +371,15 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 The edit mechanism is facilitated by `AddressBook`. It implements `AddressBook#setPerson(Person target, Person editedPerson)` which allow users to edit patient’s details in the addressbook.
 
-These operations are exposed in the `Model` interface as `Model#setPerson(Person target, Person editedPerson)`
+These operations are exposed in the `Model` interface as `Model#setPerson(Person target, Person editedPerson)`. 
+
+The `edit` feature has the following operations in `ModelManager` which implements the `Model` interface:
+
+- Model#setPerson: Replaces the given person target with `editedPerson`. target must exist in the address book. The person identity of `editedPerson` must not be the same as another existing person in the address book.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: Capitalise the 't' for 'target'

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
 * Current choice: Displays the correct error message based on the type of error made (e.g. missing fields, invalid ic format).
     * Rationale: Users will be able to learn of their error quickly and have an idea of what to edit to make the command successful.
 
+### Show feature
+
+#### Implementation
+
+The `show` mechanism is facilitated by `ModelManager`. It implements 
+- `ModelManager#setDisplayNote(Person person)` - which allows users to display
+the notes of a selected contacts on the `NoteDisplay`.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: can remove the 'a'

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+- `ModelManager#setDisplayNote(Person person)` - which allows users to display
+the notes of a selected contacts on the `NoteDisplay`. 
+- `ModelManager#clear()` -  which clears the notes in `NoteDislay`.
+
+The `ShowCommandParser` parses the user input and implements `ShowCommand#createClearCommand()` if input is an empty string, else, 
+it implements `ShowCommand#createShowCommad(IdentityCardNumberMatchesPredicate icPredicate)`. 
+
+Given below is an example usage scenario and how the `show` mechanism behaves at each step.
+
+Step 1: the user launches the application. The `AddressBook` will be initialized with the initial address book state.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: capitalise the t for the

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 2:
+
+- Scenario 1: The user executes `show T0123456A ...` to show the notes of the person in the address book with the unique identification number of `T0123456A`.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: change identification number to IC number?

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+Given below is an example usage scenario and how the `show` mechanism behaves at each step.
+
+Step 1: the user launches the application. The `AddressBook` will be initialized with the initial address book state.
+
+Step 2:
+
+- Scenario 1: The user executes `show T0123456A ...` to show the notes of the person in the address book with the unique identification number of `T0123456A`. 
+The `show` command calls `ModelManager#setDisplayNote(Person person)`, causing the modified state of the address book after the `show T0123456A ...` command executes to be displayed.
+
+- Scenario 2: The user executes `show` to clear the notes of the person in `NotesDisplay` of the address book.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: change the 'notesdisplay' to patient notes display so its standardise with the bottom

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+  The `show` command calls `ModelManager#clear()`, causing the modified state of the address book after the `show` command executes to be displayed.
+
+The following sequence diagram illustrates how a `show` operation goes through the `Logic` component and sets the Patient Note Display.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: can add the s to the note (note -> notes)

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+<puml src="diagrams/ShowSequenceDiagram.puml" alt="FindSequenceDiagram" />
+
+<box type="info" seamless>
+
+**Note:** The lifeline for `ShowCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+</box>
+
+The following activity diagram summarizes what happens when a user executes a new command:
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: execute the `show` command?

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+<div style="text-align: center;">
+    <puml src="diagrams/ShowCommandActivityDiagram.puml" width="600"/>
+</div>
+
+#### Design Considerations & Alternatives Considered 
+
+**Aspect: How to show the notes of patient on Patient Note Display.**
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: note -> notes

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+- **Alternative 1 (Current Choice):** Show patient's notes on one command using `IC_NUMBER`.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: should on be in instead?

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+    - Pros: Users are able to view specific user's notes.
+
+    - Cons: Only allow for one parameter which is `IC_NUMBER`.
+
+- **Alternative 2:** Only show the patient notes which was last searched for.
+    - Pros: Easy to implement.
+
+    - Cons: Brings inconvenience to users if they only want to view a specific user's notes.
+
+**Aspect: How to clear the notes of patient on Patient Note Display.**
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: note -> notes

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+- **Alternative 1 (Current Choice):** Clears patient's notes on Patient Note Display using one command.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: note -> notes

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -411,6 +419,103 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+    - Pros: Easy to implement. Users do not need to remember so many commands.
+
+    - Cons: The command might not be intuitive.
+
+- **Alternative 2:** Clears patient's notes on Patient Note Display with another command.
```
Comment (tengcharmaine) [inline docs/DeveloperGuide.md]: notes

**Review by tengcharmaine — APPROVED** [review]
LGTM

**Review by jiahui0309 — APPROVED** [review]
LGTM

**Pull request merged**